src/diffusers/pipelines/blip_diffusion/pipeline_blip_diffusion.py

```diff
@@ -0,0 +1,383 @@
+    def __call__(
+        self,
+        prompt,
+        reference_image,
+        source_subject_category,
+        target_subject_category,
+        latents=None,
+        guidance_scale=7.5,
+        height=512,
+        width=512,
+        seed=42,
+        num_inference_steps=50,
+        neg_prompt="",
+        prompt_strength=1.0,
+        prompt_reps=20,
+    ):
+        """
+        Function invoked when calling the pipeline for generation.
+
+        Args:
+            prompt :
+                The prompt or prompts to guide the image generation.
+            reference_image :
+                The reference image to condition the generation on.
+            source_subject_category :
+                The source subject category.
+            target_subject_category :
+                The target subject category.
+            latents :
+                Pre-generated noisy latents, sampled from a Gaussian distribution, to be used as inputs for image
+                generation. Can be used to tweak the same generation with different prompts. If not provided, a latents
+                tensor will ge generated by random sampling.
+            guidance_scale :
+                Guidance scale as defined in [Classifier-Free Diffusion Guidance](https://arxiv.org/abs/2207.12598).
+                `guidance_scale` is defined as `w` of equation 2. of [Imagen
+                Paper](https://arxiv.org/pdf/2205.11487.pdf). Guidance scale is enabled by setting `guidance_scale >
+                1`. Higher guidance scale encourages to generate images that are closely linked to the text `prompt`,
+                usually at the expense of lower image quality.
+            height :
+                The height of the generated image.
+            width :
+                The width of the generated image.
+            seed :
+                The seed to use for random generation.
+            num_inference_steps :
+                The number of denoising steps. More denoising steps usually lead to a higher quality image at the
+                expense of slower inference.
+            neg_prompt :
+                The prompt or prompts not to guide the image generation. Ignored when not using guidance (i.e., ignored
+                if `guidance_scale` is less than `1`).
+            prompt_strength :
+                The strength of the prompt. Specifies the number of times the prompt is repeated along with prompt_reps
+                to amplify the prompt.
+            prompt_reps :
+                The number of times the prompt is repeated along with prompt_strength to amplify the prompt.
+        Examples:
+
+        Returns:
+            `List[PIL.Image.Image]` : The generated images.
+        """
+
+        prompt = [self.preprocess_caption(prompt)]
+        source_subject_category = [self.preprocess_caption(source_subject_category)]
+        target_subject_category = [self.preprocess_caption(target_subject_category)]
```

Comment:
Why do we need to additionally preprocess here? I mean we don't do this for the rest of the pipelines. Could I get a brief explanation behind doing this?

Reply:
I added it to replicate the processing done in the original paper. On a closer look, it seems to just clean the prompt using basic regex, we can remove it if needed